packages/shell-bson-parser/src/index.spec.ts

```diff
@@ -448,51 +448,137 @@ describe('@mongodb-js/shell-bson-parser', function () {
           toISOString: (${newDate}).toISOString(),
           valueOf: (${newDate}.valueOf()),
        }`;
-              expect(parse(input, options)).to.deep.equal({
```

Comment:
Had to rework this test due to race conditions where the test run could span multiple milliseconds, resulting in failures when comparing the fields that have millisecond precision.